### PR TITLE
Delete stored procedures 

### LINKED ISSUE #465: Add ability to delete stored procedures

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Fixes #465

### CODE REVIEW

File: src/docdb/tree/DocDBStoredProcedureTreeItem.ts [hunk 1]
```diff
@@ -36,4 +39,19 @@ export class DocDBStoredProcedureTreeItem implements IAzureTreeItem {
...
     }
+
+    public async deleteTreeItem(_node: IAzureNode): Promise<void> {
+        const message: string = `Are you sure you want to delete collection '${this.label}' and its contents?`;
+        const result = await vscode.window.showWarningMessage(message, DialogBoxResponses.Yes, DialogBoxResponses.Cancel);
+        if (result === DialogBoxResponses.Yes) {
+            const client = getDocumentClient(this._documentEndpoint, this._masterKey, this._isEmulator);
+            await new Promise((resolve, reject) => {
+                client.deleteStoredProcedure(this.link, function (err) {
+                    err ? reject(err) : resolve();
```
Comment (PrashanthCorp) [inline src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: For a destructive operation like this, would it make sense to write the result of this operation to the output,?

Comment (StephenWeatherford) [reply src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: Don't think we're doing it for anything else like that.  Will check.

File: src/docdb/tree/DocDBStoredProcedureTreeItem.ts [hunk 2]
```diff
@@ -15,7 +18,7 @@ export class DocDBStoredProcedureTreeItem implements IAzureTreeItem {
     public static contextValue: string = "cosmosDBStoredProcedure";
     public readonly contextValue: string = DocDBStoredProcedureTreeItem.contextValue;
 
-    constructor(private _procedure: ProcedureMeta) {
+    constructor(private _documentEndpoint: string, private _masterKey: string, private _isEmulator: boolean, private _procedure: ProcedureMeta) {
```
Comment (PrashanthCorp) [inline src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: 1. Do procedures have a unique endpoint? If yes, I prefer calling it something like `storedProcEndpoint`, `ProcedureEnpoint`, or something to that effect. :) 
2. Can the variables masterKey, and isEmulator be inferred from the parent?

File: src/docdb/tree/DocDBStoredProcedureTreeItem.ts [hunk 3]
```diff
@@ -36,4 +38,19 @@ export class DocDBStoredProcedureTreeItem implements IAzureTreeItem {
             dark: path.join(__filename, '..', '..', '..', '..', '..', 'resources', 'icons', 'dark', 'Process_16x.svg')
         };
     }
+
+    public async deleteTreeItem(_node: IAzureNode): Promise<void> {
+        const message: string = `Are you sure you want to delete collection '${this.label}' and its contents?`;
```
Comment (EricJizbaMSFT) [inline src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: 'collection' -> 'stored procedure'

File: src/docdb/tree/DocDBStoredProcedureTreeItem.ts (hunk 1 above)
Comment (StephenWeatherford) [reply src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: @fiveisprime Do you want us to print to the output window for every action?  We're currently doing it for deleting accounts but nothing else I think.

File: src/docdb/tree/DocDBStoredProcedureTreeItem.ts (hunk 2 above)
Comment (StephenWeatherford) [reply src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: Might make sense, but not going to worry about it now.

File: src/docdb/tree/DocDBStoredProcedureTreeItem.ts (hunk 3 above)
Comment (StephenWeatherford) [reply src/docdb/tree/DocDBStoredProcedureTreeItem.ts]: fixed

**Pull request merged**